### PR TITLE
Fixed issue #121 (Dying to map things in infection doesn't infect you)

### PR DESCRIPTION
**Pull request by ASpoonPlaysGames**
also to prevent a crash (and for insult to injury) the round winning kill replay will show them dying to the map if they were the last human

### CODE REVIEW

**Comment by ASpoonPlaysGames** [comment]
Looking at the file differences, it seems that somebody else has done the exact same fix and beaten me to it, except they haven't done the kill replay things. In hindsight it probably wouldn't have crashed in practice, but there was potential, anyway it now will actually show the last player dying instead of the second to last so I guess that's something

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_inf.gnut [hunk 1]
```diff
@@ -59,10 +59,15 @@ void function InfectionOnPlayerKilled( entity victim, entity attacker, var damag
...
 
-	if ( attacker.IsPlayer() ) {
+	if ( attacker.IsPlayer() )
+	{
 		attacker.SetPlayerGameStat( PGS_ASSAULT_SCORE, attacker.GetPlayerGameStat( PGS_ASSAULT_SCORE ) + 1 )
 		SetRoundWinningKillReplayAttacker(attacker)
 	}
+	else
+	{
+		SetRoundWinningKillReplayAttacker(player) // everybody now gets to witness you dying to the game
```
Comment (abarichello) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_inf.gnut:69]: Variable `player` is not defined

Comment (ASpoonPlaysGames) [reply Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_inf.gnut]: yeah thats just me being a moron and forgetting to change it when i moved my functions around to match most recent version, should be fixed now

Comment (abarichello) [reply Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_inf.gnut]: So, this is supposed to trigger when the player suicides to the map right? I tried testing this by dying in out of bounds areas and the final killcam didn't show up

Comment (ASpoonPlaysGames) [reply Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_inf.gnut]: Yeah, not sure why that's not working for you, I'll test it on my computer in a sec, try and figure out where its fucking up

Comment (ASpoonPlaysGames) [reply Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_inf.gnut]: seems to work fine for me? not sure why it's not working for you atm (most recent commit at least)

Comment (abarichello) [reply Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_inf.gnut]: Still can't trigger it. I tried killing myself by falling out of bounds or running out of 'out of bounds timer' while being the last survivor

**Comment by ASpoonPlaysGames** [comment]
gonna close this pr, can't seem to get it to work properly for everybody in the lobby, unsure why. If I figure it out ill make a new one/reopen this one (can you even do that?)

**Pull request closed without merging**